homeassistant/components/tplink/config_flow.py

```diff
@@ -444,28 +567,43 @@ async def _async_reload_requires_auth_entries(self) -> None:
                     _config_entries.flow.async_abort(flow["flow_id"])
 
     @callback
-    def _async_create_entry_from_device(self, device: Device) -> ConfigFlowResult:
+    def _async_create_or_update_entry_from_device(
+        self, device: Device, *, camera_data: dict | None = None
+    ) -> ConfigFlowResult:
         """Create a config entry from a smart device."""
-        # This is only ever called after a successful device update so we know that
-        # the credential_hash is correct and should be saved.
-        self._abort_if_unique_id_configured(updates={CONF_HOST: device.host})
+        entry = None
+        if self.source == SOURCE_RECONFIGURE:
+            entry = self._get_reconfigure_entry()
+        elif self.source == SOURCE_REAUTH:
+            entry = self._get_reauth_entry()
+        if not entry:
+            self._abort_if_unique_id_configured(updates={CONF_HOST: device.host})
         data: dict[str, Any] = {
```

Comment:
```suggestion
        elif self.source == SOURCE_REAUTH:
            entry = self._get_reauth_entry()

        if not entry:
            self._abort_if_unique_id_configured(updates={CONF_HOST: device.host})

        data: dict[str, Any] = {
```
Please add some logical separation to make it easier to read.